crates/types/src/joints.rs

```diff
@@ -671,6 +671,20 @@ impl Joints<f32> {
         }
     }
 
+    pub fn from_iterator<I>(angles: I) -> Self
```

Comment:
Following Rust naming you should abbreviate to

```suggestion
    pub fn from_iter<I>(angles: I) -> Self
```

Reply:
Done